### PR TITLE
Significant cleanup of tests

### PR DESCRIPTION
**Pull request by fppt**

### CODE REVIEW

File: grakn-engine/pom.xml [hunk 1]
```diff
@@ -82,6 +96,12 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <!-- Used for creating a in memory mock Redis for testing -->
+        <dependency>
+            <groupId>ai.grakn</groupId>
+            <artifactId>redis-mock</artifactId>
+            <version>${redis-mock.version}</version>
+        </dependency>
```
Comment (aelred) [inline grakn-engine/pom.xml]: Does this need `<scope>test</scope>`?

File: grakn-engine/src/test/java/ai/grakn/engine/EngineTestHelper.java [hunk 2]
```diff
@@ -1,120 +0,0 @@
...
-        MetricRegistry metricRegistry = metricRegistry();
-        EngineID engineID = engineId();
-        TaskManager taskManager = taskManager(config, factory, jedisPool, lockProvider, engineID, metricRegistry);
-        GraknEngineStatus graknEngineStatus = graknEngineStatus();
-        ExecutorService executorService = executorService();
-        HttpHandler httpHandler = new HttpHandler(config, sparkService(), factory, metricRegistry, graknEngineStatus, taskManager, executorService);
-        return new GraknEngineServer(config, taskManager, factory, lockProvider, graknEngineStatus, redisWrapper, executorService, httpHandler, engineID);
-    }
-
-}
```
Comment (aelred) [inline grakn-engine/src/test/java/ai/grakn/engine/EngineTestHelper.java]: Ahhh deleting stuff :D

**Comment by fppt** [comment]
!rtg

**Comment by fppt** [comment]
bad change is bad

**Pull request closed without merging**